tests/functional/api.test.js

```diff
@@ -352,3 +352,41 @@ testDriver.test('api is available when sessionStorage is not', function (t, brow
     })
     .catch(fail(t))
 })
+
+testDriver.test('setUserId adds correct attribute to jserror', function (t, browser, router) {
+  let url = router.assetURL('instrumented.html', {
+    init: {
+      jserrors: {
+        harvestTimeSeconds: 2
+      }
+    },
+    scriptString: `
+    newrelic.setUserId(456);
+    newrelic.setUserId({'foo':'bar'});
+    newrelic.noticeError('fake1')
+    setTimeout(() => {  // there's a delay with errors grabbing the jsAttributes
```

Comment:
The comment? Without a `setTimeout`, what happens is that `fake1` error will have the `user123` ID. There's a race condition between `noticeError` and inner `appendJsAttribute`, in which the latter always win because it doesn't go thru the whole event-emitter system. So it looks sequential, but it really isn't; hence this way I can temporally isolate the two "sets" of tests.